### DISCUSSION TITLE
[Storage API] Should bucket id be part of the 'path' parameter?

### DISCUSSION THREAD

**Discussion by mgm1313** [supabase/supabase#1649] [Questions] [answered]
**TLDR; Should we include the actual bucket name inside the `path` parameter when uploading or updating a file?** 

I'm a bit in doubt whether we need to include the bucket name in the `path` parameter of the `from.upload()` storage api. 
When reading the [docs](https://supabase.io/docs/reference/javascript/storage-from-upload), it's listed that:

> _path_ required string
> The relative file path **including the bucket ID.** Should be of the format **bucket**/folder/subfolder/filename.png. The bucket must already exist before attempting to upload.

This also reflects in the example given on that same page:
```js
const avatarFile = event.target.files[0]
const { data, error } = await supabase
  .storage
  .from('avatars')
  .upload('public/avatar1.png', avatarFile)
```

Although I'm not quite sure by this example whether `avatars` or `public` is the actual bucket? My guess would be `avatars` is the bucket, and `public` is just a folder inside of this bucket. Which wouldn't correspondent with the format of the `path` parameter, which explicitly says to include the bucket name as first part of the string 🤔 

The [example repo of the storage api](https://github.com/supabase/supabase/blob/master/examples/nextjs-ts-user-management/components/Account.tsx) doesn't include the bucket name either, since the filePath just consists of a random string and extension:
```tsx
const user = supabase.auth.user()      
const file = event.target.files[0]      
const fileExt = file.name.split('.').pop()      
const fileName = `${session?.user.id}${Math.random()}.${fileExt}`      
const filePath = `${fileName}`

let { error: uploadError } = await supabase.storage.from(DEFAULT_AVATARS_BUCKET).upload(filePath, file)
```

**Comment by inian** [comment, marked as answer]
Hi @mgm1313, thanks for pointing this out. The docs were wrong and I updated them just now. Hope this clarifies the issue.
